clang/test/SemaCXX/cxx2b-deducing-this.cpp

```diff
@@ -1073,3 +1073,10 @@ int main() {
   return foo[]; // expected-error {{no viable overloaded operator[] for type 'Foo'}}
 }
 }
+
+namespace GH100394 {
+struct C {
+  void f(this const C);
+  void f() const ;      // ok
+};
```

Comment:
Here is a fun test 
```
struct C {
  void f(this const C);
  void f(this volatile C) ;  
};

(deprecated but should be handled nevertheless)

Reply:
@cor3ntin thanks for the additional case, I'll take a look., it seems the current implementation doesn't support it https://compiler-explorer.com/z/TYdvG1PsK

Reply:
@cor3ntin @erichkeane I've changed the normalization to omit early returns and covered this case and several others - https://compiler-explorer.com/z/TMTo5s998. Are there any additional cases that should be covered? Thanks